### PR TITLE
RPC: Tolerate unknown parameters, but with clear warning/errors

### PR DESCRIPTION
**Pull request by luke-jr**
By tolerating unknown extra rpcauth parameters (and ignoring the rpcauth), we can ensure a limited forward compatibility by not forcing users to downgrade their config file to switch back to older versions (perhaps temporarily).

To avoid any possible confusion, a message is still printed at startup, and returned as part of the error message if the user attempts to authenticate using the correct username/password.

### CODE REVIEW

File: src/httprpc.cpp [hunk 1]
```diff
@@ -256,8 +278,11 @@ static bool InitRPCAuthentication()
...
             boost::split(fields, rpcauth, boost::is_any_of(":$"));
-            if (fields.size() == 3) {
+            if (fields.size() >= MIN_RPCAUTH_VALUES) {
                 g_rpcauth.push_back(fields);
+                if (fields.size() > MAX_RPCAUTH_VALUES) {
+                    LogPrintf("Unrecognised -rpcauth parameters for username '%s'. User will not be able to authenticate.\n", SanitizeString(fields[0]));
+                }
             } else {
                 LogPrintf("Invalid -rpcauth argument.\n");
                 return false;
```
Comment (MarcoFalke) [inline src/httprpc.cpp:288]: I think you are overcomplicating this. No need for a +53-7 patch when this line could be removed in a -1+0 patch to achieve the same user experience

**Review by maflcko — CHANGES_REQUESTED** [review]

File: src/httprpc.cpp (hunk 1 above)
Comment (luke-jr) [reply src/httprpc.cpp:288]: #20550

**Comment by jnewbery** [comment]
NACK for the same reason as here: https://github.com/bitcoin/bitcoin/pull/20550#issuecomment-737820067.

We shouldn't make error handling worse in Bitcoin Core to make life easier for maintainers of downstream projects.

**Comment by MarcoFalke** [comment]
Closing for now due to controversy

**Pull request closed without merging**